subsys/bluetooth/host/smp.c

```diff
@@ -2860,18 +2860,21 @@ static uint8_t smp_pairing_req(struct bt_smp *smp, struct net_buf *buf)
 		atomic_set_bit(smp->flags, SMP_FLAG_CT2);
 	}
 
-	smp->local_dist = rsp->resp_key_dist;
-	smp->remote_dist = rsp->init_key_dist;
-
 	if ((rsp->auth_req & BT_SMP_AUTH_BONDING) &&
 	    (req->auth_req & BT_SMP_AUTH_BONDING)) {
 		atomic_set_bit(smp->flags, SMP_FLAG_BOND);
 	} else if (IS_ENABLED(CONFIG_BT_BONDING_REQUIRED)) {
 		/* Reject pairing req if not both intend to bond */
 		LOG_DBG("Bonding required");
 		return BT_SMP_ERR_UNSPECIFIED;
+	} else {
+		rsp->init_key_dist = 0;
+		rsp->resp_key_dist = 0;
```

Comment:
I'm a little confused about this. 

The spec does state, as you write in the commit message, that 
> bonding information shall not be exchanged or stored

"Bonding information" is then defined as 
> The security and identity information as defined in [Vol 3] Part H, Section 2.4.1 is also known as the bonding information

"Security information" is then defined as 
> security information (LTK, EDIV, and Rand)

But in 5.3, vol 3, Part H, 2.4.4.2 Encryption setup using LTK it also states
> The Central must have the security information (LTK, EDIV, and Rand) distributed by the Peripheral in LE legacy pairing or the LTK generated in LE Secure Connections to setup an encrypted session.

So if bonding is not supported, then we shall not exchange the security information, but the spec also states that the security information shall be exchanged to encrypt the link (as I read it anyhow). Doesn't this (wrongly) imply that if you don't support bonding, you cannot encrypt the link? :s 

Am I missing something?

Reply:
#AutoPTS autotest run zephyr GAP SM

Reply:
One will not be able to re-encrypt the link after re-connection without re-pairing. That is exactly what is no-bonding about. It is still possible to encrypt the link in the end of pairing with STK in case of legacy pairing or LTK for LESC.

Reply:
@jhedberg @alwa-nordic Any comments here? Am I misreading something, or is there a spec issue here?

Reply:
> @jhedberg @alwa-nordic Any comments here? Am I misreading something, or is there a spec issue here?

I'm not sure I completely understand your concern. The situation is like @seko-nordic described in his earlier comment. You'll be able to encrypt, but just for the same connection that you run the SMP procedure over. That's how SMP with no-bonding is expected to behave.

Reply:
I agree, but my concern is not with this PR, but the text in the specifications. To me it seems that the definitions of bonding information and security information don't match up.